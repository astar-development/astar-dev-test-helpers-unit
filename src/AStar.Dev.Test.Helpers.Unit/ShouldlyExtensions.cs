using AStar.Dev.Utilities;
using Shouldly;

namespace AStar.Dev.Test.Helpers.Unit;

/// <summary>
///     The <see cref="ShouldlyExtensions" /> class containing the current extensions for the Shouldly package
/// </summary>
public static class ShouldlyExtensions
{
    /// <summary>
    ///     The ShouldMatchMaskedApproved method abstracts the original ShouldMatchApproved and adds masking for ISO Dates and GUIDs within the object being compared
    /// </summary>
    /// <param name="source">The object to check to the approved version</param>
    /// <typeparam name="T">The type of the object being compared</typeparam>
    public static void ShouldMatchMaskedApproved<T>(this T source)
        => source.ToJson().ShouldMatchApproved(options
                                                   => options
                                                      .WithScrubber(input => ShouldlyExtensionsRegex.RegexIsoDate().Replace(input, "<date>"))
                                                      .WithScrubber(input => ShouldlyExtensionsRegex.RegexGuid().Replace(input, "<GUID>")));
}