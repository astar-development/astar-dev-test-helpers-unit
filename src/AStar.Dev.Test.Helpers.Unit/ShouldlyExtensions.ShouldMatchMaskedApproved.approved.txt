{
  "irrelevantDate": "<date>",
  "irrelevantGuid": "<GUID>",
  "irrelevantString": "Irrelevant string"
}